internal/auth/authorizations.go

```diff
@@ -78,19 +88,38 @@ func (a *State) IsExpired() bool {
 }
 
 func QueryAnyAuthd(authd []string, remoteIP string, tlsEnabled bool, commonName string, authSecret string,
-	connectTimeout time.Duration, requestTimeout time.Duration) (*State, error) {
+	authConfig *AuthConfig) (*State, error) {
 	start := rand.Int()
 	n := len(authd)
 	for i := 0; i < n; i++ {
 		a := authd[(i+start)%n]
-		authState, err := QueryAuthd(a, remoteIP, tlsEnabled, commonName, authSecret, connectTimeout, requestTimeout)
+		authState, err := QueryAuthd(a, remoteIP, tlsEnabled, commonName, authSecret, authConfig.ConnectTimeout, authConfig.RequestTimeout)
 		if err != nil {
 			log.Printf("Error: failed auth against %s %s", a, err)
 			continue
 		}
 		return authState, nil
 	}
-	return nil, errors.New("Unable to access auth server")
+
+	if authConfig.FailClosed {
+		log.Printf("Unable to access an auth server, failing closed and allowing all authorizations")
```

Comment:
i'm not sure how the logging semantics work for nsq, but showing this as a warning or error might be good?

Reply:
nsqd has a special logging function with levels. For example, in `nsqd/nsqd.go`, it calls:

```go
			n.logf(LOG_WARN, "skipping creation of invalid topic %s", t.Name)
```

Internal modules do not have access to the NSQD struct, so for some of them, it passes the log function when it initializes the internal module, for example:

```go
		exitFunc(protocol.TCPServer(n.tcpListener, tcpServer, n.logf))
```

```go
func TCPServer(listener net.Listener, handler TCPHandler, logf lg.AppLogFunc) error {
	logf(lg.INFO, "TCP: listening on %s", listener.Addr())
        ...
```

This internal/authorizations module has thus far not had the log function passed because it only returned errors which would be logged by another module.